sql/exec: TestSortRandomized failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/33f9a3c2c5a0d2b756a8827daa74285b1aece401

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestSortRandomized PKG=github.com/cockroachdb/cockroach/pkg/sql/exec TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1331313&tab=buildLog

```
        --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=16/sel=true (0.00s)
            sort_test.go:186: for input [[1197 1712 974] [504 1957 1180] [1500 174 974] [725 24 870] [1159 902 15] [1207 1135 1885] [675 1984 1129] [1446 1739 445]]:
                expected [[1159 902 15] [1446 1739 445] [725 24 870] [1197 1712 974]], actual [[1159 902 15] [1446 1739 445] [725 24 870] [1500 174 974]]
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=1024/sel=true
        --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=1024/sel=true (0.00s)
            sort_test.go:186: for input [[1197 1712 974] [504 1957 1180] [1500 174 974] [725 24 870] [1159 902 15] [1207 1135 1885] [675 1984 1129] [1446 1739 445]]:
                expected [[1159 902 15] [1446 1739 445] [725 24 870] [1197 1712 974]], actual [[1159 902 15] [1446 1739 445] [725 24 870] [1500 174 974]]
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true
    --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true (0.00s)
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=2/sel=true
        --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=2/sel=true (0.00s)
            sort_test.go:186: for input [[1197 1712 974] [504 1957 1180] [1500 174 974] [725 24 870] [1159 902 15] [1207 1135 1885] [675 1984 1129] [1446 1739 445]]:
                expected [[1159 902 15] [1446 1739 445] [725 24 870] [1197 1712 974]], actual [[1159 902 15] [1446 1739 445] [725 24 870] [1500 174 974]]
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=3/sel=false
        --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=3/sel=false (0.00s)
            sort_test.go:186: for input [[1197 1712 974] [504 1957 1180] [1500 174 974] [725 24 870] [1159 902 15] [1207 1135 1885] [675 1984 1129] [1446 1739 445]]:
                expected [[1159 902 15] [1446 1739 445] [725 24 870] [1197 1712 974]], actual [[1159 902 15] [1446 1739 445] [725 24 870] [1500 174 974]]
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=2/sel=false
        --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=2/sel=false (0.00s)
            sort_test.go:186: for input [[1197 1712 974] [504 1957 1180] [1500 174 974] [725 24 870] [1159 902 15] [1207 1135 1885] [675 1984 1129] [1446 1739 445]]:
                expected [[1159 902 15] [1446 1739 445] [725 24 870] [1197 1712 974]], actual [[1159 902 15] [1446 1739 445] [725 24 870] [1500 174 974]]
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=1024/sel=false
        --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=1024/sel=false (0.00s)
            sort_test.go:186: for input [[1197 1712 974] [504 1957 1180] [1500 174 974] [725 24 870] [1159 902 15] [1207 1135 1885] [675 1984 1129] [1446 1739 445]]:
                expected [[1159 902 15] [1446 1739 445] [725 24 870] [1197 1712 974]], actual [[1159 902 15] [1446 1739 445] [725 24 870] [1500 174 974]]
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=1/sel=false
        --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=1/sel=false (0.00s)
            sort_test.go:186: for input [[1197 1712 974] [504 1957 1180] [1500 174 974] [725 24 870] [1159 902 15] [1207 1135 1885] [675 1984 1129] [1446 1739 445]]:
                expected [[1159 902 15] [1446 1739 445] [725 24 870] [1197 1712 974]], actual [[1159 902 15] [1446 1739 445] [725 24 870] [1500 174 974]]
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=1/sel=true
        --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=1/sel=true (0.00s)
            sort_test.go:186: for input [[1197 1712 974] [504 1957 1180] [1500 174 974] [725 24 870] [1159 902 15] [1207 1135 1885] [675 1984 1129] [1446 1739 445]]:
                expected [[1159 902 15] [1446 1739 445] [725 24 870] [1197 1712 974]], actual [[1159 902 15] [1446 1739 445] [725 24 870] [1500 174 974]]

```

SHA: https://github.com/cockroachdb/cockroach/commits/de90b5d84c9e5d2ef562ebfa3d4a1bdc0e834e10

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestSortRandomized PKG=github.com/cockroachdb/cockroach/pkg/sql/exec TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1332594&tab=buildLog

```
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=1024/sel=false (0.00s)
            sort_test.go:186: for input [[1668 1447] [1602 1349] [1301 599] [796 550] [304 1151] [553 1149] [1602 1222] [1510 1896]]:
                expected [[1668 1447] [1602 1349] [1602 1222] [1510 1896]], actual [[1668 1447] [1602 1222] [1602 1349] [1510 1896]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=1024/sel=true
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=1024/sel=true (0.00s)
            sort_test.go:186: for input [[1668 1447] [1602 1349] [1301 599] [796 550] [304 1151] [553 1149] [1602 1222] [1510 1896]]:
                expected [[1668 1447] [1602 1349] [1602 1222] [1510 1896]], actual [[1668 1447] [1602 1222] [1602 1349] [1510 1896]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=2/sel=false
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=2/sel=false (0.00s)
            sort_test.go:186: for input [[1668 1447] [1602 1349] [1301 599] [796 550] [304 1151] [553 1149] [1602 1222] [1510 1896]]:
                expected [[1668 1447] [1602 1349] [1602 1222] [1510 1896]], actual [[1668 1447] [1602 1222] [1602 1349] [1510 1896]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=2/sel=true
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=2/sel=true (0.00s)
            sort_test.go:186: for input [[1668 1447] [1602 1349] [1301 599] [796 550] [304 1151] [553 1149] [1602 1222] [1510 1896]]:
                expected [[1668 1447] [1602 1349] [1602 1222] [1510 1896]], actual [[1668 1447] [1602 1222] [1602 1349] [1510 1896]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=3/sel=true
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=3/sel=true (0.00s)
            sort_test.go:186: for input [[1668 1447] [1602 1349] [1301 599] [796 550] [304 1151] [553 1149] [1602 1222] [1510 1896]]:
                expected [[1668 1447] [1602 1349] [1602 1222] [1510 1896]], actual [[1668 1447] [1602 1222] [1602 1349] [1510 1896]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=16/sel=true
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=16/sel=true (0.00s)
            sort_test.go:186: for input [[1668 1447] [1602 1349] [1301 599] [796 550] [304 1151] [553 1149] [1602 1222] [1510 1896]]:
                expected [[1668 1447] [1602 1349] [1602 1222] [1510 1896]], actual [[1668 1447] [1602 1222] [1602 1349] [1510 1896]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true
    --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true (0.00s)
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=1/sel=false
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=1/sel=false (0.00s)
            sort_test.go:186: for input [[1668 1447] [1602 1349] [1301 599] [796 550] [304 1151] [553 1149] [1602 1222] [1510 1896]]:
                expected [[1668 1447] [1602 1349] [1602 1222] [1510 1896]], actual [[1668 1447] [1602 1222] [1602 1349] [1510 1896]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=1/sel=true
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=1/sel=true (0.00s)
            sort_test.go:186: for input [[1668 1447] [1602 1349] [1301 599] [796 550] [304 1151] [553 1149] [1602 1222] [1510 1896]]:
                expected [[1668 1447] [1602 1349] [1602 1222] [1510 1896]], actual [[1668 1447] [1602 1222] [1602 1349] [1510 1896]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=3/sel=false
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=3/sel=false (0.00s)
            sort_test.go:186: for input [[1668 1447] [1602 1349] [1301 599] [796 550] [304 1151] [553 1149] [1602 1222] [1510 1896]]:
                expected [[1668 1447] [1602 1349] [1602 1222] [1510 1896]], actual [[1668 1447] [1602 1222] [1602 1349] [1510 1896]]

```

SHA: https://github.com/cockroachdb/cockroach/commits/3b63648f905715e6c0b055fe9acac9c5b8206196

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestSortRandomized PKG=github.com/cockroachdb/cockroach/pkg/sql/exec TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1333780&tab=buildLog

```
=== RUN   TestSortRandomized/nCols=4/nOrderingCols=1/topK=true/batchSize=16/sel=true
        --- FAIL: TestSortRandomized/nCols=4/nOrderingCols=1/topK=true/batchSize=16/sel=true (0.00s)
            sort_test.go:186: for input [[1231 640 1428 784] [1838 60 679 2000] [1653 1609 1879 1186] [1217 1485 660 882] [1105 56 1390 2000] [539 350 1631 1661] [355 293 1564 1352] [1074 1609 763 844]]:
                expected [[1838 60 679 2000] [1105 56 1390 2000] [539 350 1631 1661] [355 293 1564 1352]], actual [[1105 56 1390 2000] [1838 60 679 2000] [539 350 1631 1661] [355 293 1564 1352]]
=== RUN   TestSortRandomized/nCols=4/nOrderingCols=1/topK=true/batchSize=1024/sel=false
        --- FAIL: TestSortRandomized/nCols=4/nOrderingCols=1/topK=true/batchSize=1024/sel=false (0.00s)
            sort_test.go:186: for input [[1231 640 1428 784] [1838 60 679 2000] [1653 1609 1879 1186] [1217 1485 660 882] [1105 56 1390 2000] [539 350 1631 1661] [355 293 1564 1352] [1074 1609 763 844]]:
                expected [[1838 60 679 2000] [1105 56 1390 2000] [539 350 1631 1661] [355 293 1564 1352]], actual [[1105 56 1390 2000] [1838 60 679 2000] [539 350 1631 1661] [355 293 1564 1352]]
=== RUN   TestSortRandomized/nCols=4/nOrderingCols=1/topK=true/batchSize=1024/sel=true
        --- FAIL: TestSortRandomized/nCols=4/nOrderingCols=1/topK=true/batchSize=1024/sel=true (0.00s)
            sort_test.go:186: for input [[1231 640 1428 784] [1838 60 679 2000] [1653 1609 1879 1186] [1217 1485 660 882] [1105 56 1390 2000] [539 350 1631 1661] [355 293 1564 1352] [1074 1609 763 844]]:
                expected [[1838 60 679 2000] [1105 56 1390 2000] [539 350 1631 1661] [355 293 1564 1352]], actual [[1105 56 1390 2000] [1838 60 679 2000] [539 350 1631 1661] [355 293 1564 1352]]
=== RUN   TestSortRandomized/nCols=4/nOrderingCols=1/topK=true/batchSize=1/sel=true
        --- FAIL: TestSortRandomized/nCols=4/nOrderingCols=1/topK=true/batchSize=1/sel=true (0.01s)
            sort_test.go:186: for input [[1231 640 1428 784] [1838 60 679 2000] [1653 1609 1879 1186] [1217 1485 660 882] [1105 56 1390 2000] [539 350 1631 1661] [355 293 1564 1352] [1074 1609 763 844]]:
                expected [[1838 60 679 2000] [1105 56 1390 2000] [539 350 1631 1661] [355 293 1564 1352]], actual [[1105 56 1390 2000] [1838 60 679 2000] [539 350 1631 1661] [355 293 1564 1352]]
=== RUN   TestSortRandomized/nCols=4/nOrderingCols=1/topK=true/batchSize=2/sel=false
        --- FAIL: TestSortRandomized/nCols=4/nOrderingCols=1/topK=true/batchSize=2/sel=false (0.00s)
            sort_test.go:186: for input [[1231 640 1428 784] [1838 60 679 2000] [1653 1609 1879 1186] [1217 1485 660 882] [1105 56 1390 2000] [539 350 1631 1661] [355 293 1564 1352] [1074 1609 763 844]]:
                expected [[1838 60 679 2000] [1105 56 1390 2000] [539 350 1631 1661] [355 293 1564 1352]], actual [[1105 56 1390 2000] [1838 60 679 2000] [539 350 1631 1661] [355 293 1564 1352]]
=== RUN   TestSortRandomized/nCols=4/nOrderingCols=1/topK=true/batchSize=3/sel=false
        --- FAIL: TestSortRandomized/nCols=4/nOrderingCols=1/topK=true/batchSize=3/sel=false (0.00s)
            sort_test.go:186: for input [[1231 640 1428 784] [1838 60 679 2000] [1653 1609 1879 1186] [1217 1485 660 882] [1105 56 1390 2000] [539 350 1631 1661] [355 293 1564 1352] [1074 1609 763 844]]:
                expected [[1838 60 679 2000] [1105 56 1390 2000] [539 350 1631 1661] [355 293 1564 1352]], actual [[1105 56 1390 2000] [1838 60 679 2000] [539 350 1631 1661] [355 293 1564 1352]]
=== RUN   TestSortRandomized/nCols=4/nOrderingCols=1/topK=true/batchSize=16/sel=false
        --- FAIL: TestSortRandomized/nCols=4/nOrderingCols=1/topK=true/batchSize=16/sel=false (0.00s)
            sort_test.go:186: for input [[1231 640 1428 784] [1838 60 679 2000] [1653 1609 1879 1186] [1217 1485 660 882] [1105 56 1390 2000] [539 350 1631 1661] [355 293 1564 1352] [1074 1609 763 844]]:
                expected [[1838 60 679 2000] [1105 56 1390 2000] [539 350 1631 1661] [355 293 1564 1352]], actual [[1105 56 1390 2000] [1838 60 679 2000] [539 350 1631 1661] [355 293 1564 1352]]

```

SHA: https://github.com/cockroachdb/cockroach/commits/254f7a8aab8d2f32315c879e0cc946429883e9e6

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestSortRandomized PKG=github.com/cockroachdb/cockroach/pkg/sql/exec TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1338185&tab=buildLog

```
            sort_test.go:186: for input [[1473 1309 1167] [345 1642 1745] [397 1364 278] [786 493 706] [2017 1285 1211] [179 1043 1813] [1935 712 1211] [1641 1596 1107]]:
                expected [[179 1043 1813] [345 1642 1745] [1935 712 1211] [2017 1285 1211]], actual [[179 1043 1813] [345 1642 1745] [2017 1285 1211] [1935 712 1211]]
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true
    --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true (0.00s)
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=1/sel=false
        --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=1/sel=false (0.00s)
            sort_test.go:186: for input [[1473 1309 1167] [345 1642 1745] [397 1364 278] [786 493 706] [2017 1285 1211] [179 1043 1813] [1935 712 1211] [1641 1596 1107]]:
                expected [[179 1043 1813] [345 1642 1745] [1935 712 1211] [2017 1285 1211]], actual [[179 1043 1813] [345 1642 1745] [2017 1285 1211] [1935 712 1211]]
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=1/sel=true
        --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=1/sel=true (0.00s)
            sort_test.go:186: for input [[1473 1309 1167] [345 1642 1745] [397 1364 278] [786 493 706] [2017 1285 1211] [179 1043 1813] [1935 712 1211] [1641 1596 1107]]:
                expected [[179 1043 1813] [345 1642 1745] [1935 712 1211] [2017 1285 1211]], actual [[179 1043 1813] [345 1642 1745] [2017 1285 1211] [1935 712 1211]]
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=2/sel=true
        --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=2/sel=true (0.00s)
            sort_test.go:186: for input [[1473 1309 1167] [345 1642 1745] [397 1364 278] [786 493 706] [2017 1285 1211] [179 1043 1813] [1935 712 1211] [1641 1596 1107]]:
                expected [[179 1043 1813] [345 1642 1745] [1935 712 1211] [2017 1285 1211]], actual [[179 1043 1813] [345 1642 1745] [2017 1285 1211] [1935 712 1211]]
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=1024/sel=false
        --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=1024/sel=false (0.00s)
            sort_test.go:186: for input [[1473 1309 1167] [345 1642 1745] [397 1364 278] [786 493 706] [2017 1285 1211] [179 1043 1813] [1935 712 1211] [1641 1596 1107]]:
                expected [[179 1043 1813] [345 1642 1745] [1935 712 1211] [2017 1285 1211]], actual [[179 1043 1813] [345 1642 1745] [2017 1285 1211] [1935 712 1211]]
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=2/sel=false
        --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=2/sel=false (0.00s)
            sort_test.go:186: for input [[1473 1309 1167] [345 1642 1745] [397 1364 278] [786 493 706] [2017 1285 1211] [179 1043 1813] [1935 712 1211] [1641 1596 1107]]:
                expected [[179 1043 1813] [345 1642 1745] [1935 712 1211] [2017 1285 1211]], actual [[179 1043 1813] [345 1642 1745] [2017 1285 1211] [1935 712 1211]]
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=3/sel=false
        --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=3/sel=false (0.00s)
            sort_test.go:186: for input [[1473 1309 1167] [345 1642 1745] [397 1364 278] [786 493 706] [2017 1285 1211] [179 1043 1813] [1935 712 1211] [1641 1596 1107]]:
                expected [[179 1043 1813] [345 1642 1745] [1935 712 1211] [2017 1285 1211]], actual [[179 1043 1813] [345 1642 1745] [2017 1285 1211] [1935 712 1211]]
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=3/sel=true
        --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=3/sel=true (0.00s)
            sort_test.go:186: for input [[1473 1309 1167] [345 1642 1745] [397 1364 278] [786 493 706] [2017 1285 1211] [179 1043 1813] [1935 712 1211] [1641 1596 1107]]:
                expected [[179 1043 1813] [345 1642 1745] [1935 712 1211] [2017 1285 1211]], actual [[179 1043 1813] [345 1642 1745] [2017 1285 1211] [1935 712 1211]]

```

SHA: https://github.com/cockroachdb/cockroach/commits/254f7a8aab8d2f32315c879e0cc946429883e9e6

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestSortRandomized PKG=github.com/cockroachdb/cockroach/pkg/sql/exec TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1338186&tab=buildLog

```
        --- FAIL: TestSortRandomized/nCols=4/nOrderingCols=1/topK=true/batchSize=3/sel=true (0.00s)
            sort_test.go:186: for input [[1183 412 1826 370] [1870 584 356 1583] [227 1782 388 59] [1183 1962 544 497] [581 1943 1774 880] [1045 1152 1902 1944] [1242 1400 1091 613] [1947 1211 962 128]]:
                expected [[227 1782 388 59] [581 1943 1774 880] [1045 1152 1902 1944] [1183 412 1826 370]], actual [[227 1782 388 59] [581 1943 1774 880] [1045 1152 1902 1944] [1183 1962 544 497]]
=== RUN   TestSortRandomized/nCols=4/nOrderingCols=1/topK=true/batchSize=16/sel=false
        --- FAIL: TestSortRandomized/nCols=4/nOrderingCols=1/topK=true/batchSize=16/sel=false (0.00s)
            sort_test.go:186: for input [[1183 412 1826 370] [1870 584 356 1583] [227 1782 388 59] [1183 1962 544 497] [581 1943 1774 880] [1045 1152 1902 1944] [1242 1400 1091 613] [1947 1211 962 128]]:
                expected [[227 1782 388 59] [581 1943 1774 880] [1045 1152 1902 1944] [1183 412 1826 370]], actual [[227 1782 388 59] [581 1943 1774 880] [1045 1152 1902 1944] [1183 1962 544 497]]
=== RUN   TestSortRandomized/nCols=4/nOrderingCols=1/topK=true/batchSize=16/sel=true
        --- FAIL: TestSortRandomized/nCols=4/nOrderingCols=1/topK=true/batchSize=16/sel=true (0.00s)
            sort_test.go:186: for input [[1183 412 1826 370] [1870 584 356 1583] [227 1782 388 59] [1183 1962 544 497] [581 1943 1774 880] [1045 1152 1902 1944] [1242 1400 1091 613] [1947 1211 962 128]]:
                expected [[227 1782 388 59] [581 1943 1774 880] [1045 1152 1902 1944] [1183 412 1826 370]], actual [[227 1782 388 59] [581 1943 1774 880] [1045 1152 1902 1944] [1183 1962 544 497]]
=== RUN   TestSortRandomized/nCols=4/nOrderingCols=1/topK=true/batchSize=1024/sel=true
        --- FAIL: TestSortRandomized/nCols=4/nOrderingCols=1/topK=true/batchSize=1024/sel=true (0.00s)
            sort_test.go:186: for input [[1183 412 1826 370] [1870 584 356 1583] [227 1782 388 59] [1183 1962 544 497] [581 1943 1774 880] [1045 1152 1902 1944] [1242 1400 1091 613] [1947 1211 962 128]]:
                expected [[227 1782 388 59] [581 1943 1774 880] [1045 1152 1902 1944] [1183 412 1826 370]], actual [[227 1782 388 59] [581 1943 1774 880] [1045 1152 1902 1944] [1183 1962 544 497]]
=== RUN   TestSortRandomized/nCols=4/nOrderingCols=1/topK=true/batchSize=1/sel=false
        --- FAIL: TestSortRandomized/nCols=4/nOrderingCols=1/topK=true/batchSize=1/sel=false (0.00s)
            sort_test.go:186: for input [[1183 412 1826 370] [1870 584 356 1583] [227 1782 388 59] [1183 1962 544 497] [581 1943 1774 880] [1045 1152 1902 1944] [1242 1400 1091 613] [1947 1211 962 128]]:
                expected [[227 1782 388 59] [581 1943 1774 880] [1045 1152 1902 1944] [1183 412 1826 370]], actual [[227 1782 388 59] [581 1943 1774 880] [1045 1152 1902 1944] [1183 1962 544 497]]
=== RUN   TestSortRandomized/nCols=4/nOrderingCols=1/topK=true
    --- FAIL: TestSortRandomized/nCols=4/nOrderingCols=1/topK=true (0.02s)
=== RUN   TestSortRandomized/nCols=4/nOrderingCols=1/topK=true/batchSize=1024/sel=false
        --- FAIL: TestSortRandomized/nCols=4/nOrderingCols=1/topK=true/batchSize=1024/sel=false (0.00s)
            sort_test.go:186: for input [[1183 412 1826 370] [1870 584 356 1583] [227 1782 388 59] [1183 1962 544 497] [581 1943 1774 880] [1045 1152 1902 1944] [1242 1400 1091 613] [1947 1211 962 128]]:
                expected [[227 1782 388 59] [581 1943 1774 880] [1045 1152 1902 1944] [1183 412 1826 370]], actual [[227 1782 388 59] [581 1943 1774 880] [1045 1152 1902 1944] [1183 1962 544 497]]
=== RUN   TestSortRandomized/nCols=4/nOrderingCols=1/topK=true/batchSize=2/sel=true
        --- FAIL: TestSortRandomized/nCols=4/nOrderingCols=1/topK=true/batchSize=2/sel=true (0.00s)
            sort_test.go:186: for input [[1183 412 1826 370] [1870 584 356 1583] [227 1782 388 59] [1183 1962 544 497] [581 1943 1774 880] [1045 1152 1902 1944] [1242 1400 1091 613] [1947 1211 962 128]]:
                expected [[227 1782 388 59] [581 1943 1774 880] [1045 1152 1902 1944] [1183 412 1826 370]], actual [[227 1782 388 59] [581 1943 1774 880] [1045 1152 1902 1944] [1183 1962 544 497]]

```

SHA: https://github.com/cockroachdb/cockroach/commits/db6d4425d65bdb027624972ccb19d7aad0bc57cc

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestSortRandomized PKG=github.com/cockroachdb/cockroach/pkg/sql/exec TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1340423&tab=buildLog

```
        --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=1024/sel=false (0.00s)
            sort_test.go:186: for input [[1856 1894 1964] [375 1669 467] [482 1329 467] [410 1207 1173] [424 1668 348] [1243 975 469] [1326 995 1379] [196 1703 1207]]:
                expected [[424 1668 348] [375 1669 467] [482 1329 467] [1243 975 469]], actual [[424 1668 348] [482 1329 467] [375 1669 467] [1243 975 469]]
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=1/sel=false
        --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=1/sel=false (0.00s)
            sort_test.go:186: for input [[1856 1894 1964] [375 1669 467] [482 1329 467] [410 1207 1173] [424 1668 348] [1243 975 469] [1326 995 1379] [196 1703 1207]]:
                expected [[424 1668 348] [375 1669 467] [482 1329 467] [1243 975 469]], actual [[424 1668 348] [482 1329 467] [375 1669 467] [1243 975 469]]
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=2/sel=true
        --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=2/sel=true (0.00s)
            sort_test.go:186: for input [[1856 1894 1964] [375 1669 467] [482 1329 467] [410 1207 1173] [424 1668 348] [1243 975 469] [1326 995 1379] [196 1703 1207]]:
                expected [[424 1668 348] [375 1669 467] [482 1329 467] [1243 975 469]], actual [[424 1668 348] [482 1329 467] [375 1669 467] [1243 975 469]]
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=3/sel=true
        --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=3/sel=true (0.00s)
            sort_test.go:186: for input [[1856 1894 1964] [375 1669 467] [482 1329 467] [410 1207 1173] [424 1668 348] [1243 975 469] [1326 995 1379] [196 1703 1207]]:
                expected [[424 1668 348] [375 1669 467] [482 1329 467] [1243 975 469]], actual [[424 1668 348] [482 1329 467] [375 1669 467] [1243 975 469]]
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=16/sel=false
        --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=16/sel=false (0.00s)
            sort_test.go:186: for input [[1856 1894 1964] [375 1669 467] [482 1329 467] [410 1207 1173] [424 1668 348] [1243 975 469] [1326 995 1379] [196 1703 1207]]:
                expected [[424 1668 348] [375 1669 467] [482 1329 467] [1243 975 469]], actual [[424 1668 348] [482 1329 467] [375 1669 467] [1243 975 469]]
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=1024/sel=true
        --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=1024/sel=true (0.00s)
            sort_test.go:186: for input [[1856 1894 1964] [375 1669 467] [482 1329 467] [410 1207 1173] [424 1668 348] [1243 975 469] [1326 995 1379] [196 1703 1207]]:
                expected [[424 1668 348] [375 1669 467] [482 1329 467] [1243 975 469]], actual [[424 1668 348] [482 1329 467] [375 1669 467] [1243 975 469]]
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true
    --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true (0.00s)
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=1/sel=true
        --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=1/sel=true (0.00s)
            sort_test.go:186: for input [[1856 1894 1964] [375 1669 467] [482 1329 467] [410 1207 1173] [424 1668 348] [1243 975 469] [1326 995 1379] [196 1703 1207]]:
                expected [[424 1668 348] [375 1669 467] [482 1329 467] [1243 975 469]], actual [[424 1668 348] [482 1329 467] [375 1669 467] [1243 975 469]]
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=2/sel=false
        --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=2/sel=false (0.00s)
            sort_test.go:186: for input [[1856 1894 1964] [375 1669 467] [482 1329 467] [410 1207 1173] [424 1668 348] [1243 975 469] [1326 995 1379] [196 1703 1207]]:
                expected [[424 1668 348] [375 1669 467] [482 1329 467] [1243 975 469]], actual [[424 1668 348] [482 1329 467] [375 1669 467] [1243 975 469]]

```

SHA: https://github.com/cockroachdb/cockroach/commits/db6d4425d65bdb027624972ccb19d7aad0bc57cc

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestSortRandomized PKG=github.com/cockroachdb/cockroach/pkg/sql/exec TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1340424&tab=buildLog

```
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=1024/sel=false
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=1024/sel=false (0.00s)
            sort_test.go:186: for input [[471 1982] [848 1581] [1122 249] [766 1631] [854 292] [226 1889] [854 1135] [1235 1786]]:
                expected [[1235 1786] [1122 249] [854 1135] [854 292]], actual [[1235 1786] [1122 249] [854 292] [854 1135]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=1024/sel=true
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=1024/sel=true (0.00s)
            sort_test.go:186: for input [[471 1982] [848 1581] [1122 249] [766 1631] [854 292] [226 1889] [854 1135] [1235 1786]]:
                expected [[1235 1786] [1122 249] [854 1135] [854 292]], actual [[1235 1786] [1122 249] [854 292] [854 1135]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true
    --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true (0.01s)
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=1/sel=true
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=1/sel=true (0.00s)
            sort_test.go:186: for input [[471 1982] [848 1581] [1122 249] [766 1631] [854 292] [226 1889] [854 1135] [1235 1786]]:
                expected [[1235 1786] [1122 249] [854 1135] [854 292]], actual [[1235 1786] [1122 249] [854 292] [854 1135]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=2/sel=false
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=2/sel=false (0.00s)
            sort_test.go:186: for input [[471 1982] [848 1581] [1122 249] [766 1631] [854 292] [226 1889] [854 1135] [1235 1786]]:
                expected [[1235 1786] [1122 249] [854 1135] [854 292]], actual [[1235 1786] [1122 249] [854 292] [854 1135]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=2/sel=true
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=2/sel=true (0.00s)
            sort_test.go:186: for input [[471 1982] [848 1581] [1122 249] [766 1631] [854 292] [226 1889] [854 1135] [1235 1786]]:
                expected [[1235 1786] [1122 249] [854 1135] [854 292]], actual [[1235 1786] [1122 249] [854 292] [854 1135]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=3/sel=false
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=3/sel=false (0.00s)
            sort_test.go:186: for input [[471 1982] [848 1581] [1122 249] [766 1631] [854 292] [226 1889] [854 1135] [1235 1786]]:
                expected [[1235 1786] [1122 249] [854 1135] [854 292]], actual [[1235 1786] [1122 249] [854 292] [854 1135]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=1/sel=false
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=1/sel=false (0.00s)
            sort_test.go:186: for input [[471 1982] [848 1581] [1122 249] [766 1631] [854 292] [226 1889] [854 1135] [1235 1786]]:
                expected [[1235 1786] [1122 249] [854 1135] [854 292]], actual [[1235 1786] [1122 249] [854 292] [854 1135]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=16/sel=false
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=16/sel=false (0.00s)
            sort_test.go:186: for input [[471 1982] [848 1581] [1122 249] [766 1631] [854 292] [226 1889] [854 1135] [1235 1786]]:
                expected [[1235 1786] [1122 249] [854 1135] [854 292]], actual [[1235 1786] [1122 249] [854 292] [854 1135]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=16/sel=true
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=16/sel=true (0.00s)
            sort_test.go:186: for input [[471 1982] [848 1581] [1122 249] [766 1631] [854 292] [226 1889] [854 1135] [1235 1786]]:
                expected [[1235 1786] [1122 249] [854 1135] [854 292]], actual [[1235 1786] [1122 249] [854 292] [854 1135]]

```

SHA: https://github.com/cockroachdb/cockroach/commits/5c0b1644f9f9fe65bfb8cf3f7a5af2595bd859a8

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestSortRandomized PKG=github.com/cockroachdb/cockroach/pkg/sql/exec TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1341476&tab=buildLog

```
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=1024/sel=true
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=1024/sel=true (0.00s)
            sort_test.go:186: for input [[1740 1699] [129 1164] [1507 1218] [817 1127] [1550 189] [207 754] [1273 1489] [1273 274]]:
                expected [[1740 1699] [1550 189] [1507 1218] [1273 274]], actual [[1740 1699] [1550 189] [1507 1218] [1273 1489]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=1/sel=false
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=1/sel=false (0.00s)
            sort_test.go:186: for input [[1740 1699] [129 1164] [1507 1218] [817 1127] [1550 189] [207 754] [1273 1489] [1273 274]]:
                expected [[1740 1699] [1550 189] [1507 1218] [1273 274]], actual [[1740 1699] [1550 189] [1507 1218] [1273 1489]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=2/sel=true
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=2/sel=true (0.00s)
            sort_test.go:186: for input [[1740 1699] [129 1164] [1507 1218] [817 1127] [1550 189] [207 754] [1273 1489] [1273 274]]:
                expected [[1740 1699] [1550 189] [1507 1218] [1273 274]], actual [[1740 1699] [1550 189] [1507 1218] [1273 1489]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=3/sel=false
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=3/sel=false (0.00s)
            sort_test.go:186: for input [[1740 1699] [129 1164] [1507 1218] [817 1127] [1550 189] [207 754] [1273 1489] [1273 274]]:
                expected [[1740 1699] [1550 189] [1507 1218] [1273 274]], actual [[1740 1699] [1550 189] [1507 1218] [1273 1489]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=3/sel=true
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=3/sel=true (0.00s)
            sort_test.go:186: for input [[1740 1699] [129 1164] [1507 1218] [817 1127] [1550 189] [207 754] [1273 1489] [1273 274]]:
                expected [[1740 1699] [1550 189] [1507 1218] [1273 274]], actual [[1740 1699] [1550 189] [1507 1218] [1273 1489]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=16/sel=false
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=16/sel=false (0.00s)
            sort_test.go:186: for input [[1740 1699] [129 1164] [1507 1218] [817 1127] [1550 189] [207 754] [1273 1489] [1273 274]]:
                expected [[1740 1699] [1550 189] [1507 1218] [1273 274]], actual [[1740 1699] [1550 189] [1507 1218] [1273 1489]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=1/sel=true
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=1/sel=true (0.00s)
            sort_test.go:186: for input [[1740 1699] [129 1164] [1507 1218] [817 1127] [1550 189] [207 754] [1273 1489] [1273 274]]:
                expected [[1740 1699] [1550 189] [1507 1218] [1273 274]], actual [[1740 1699] [1550 189] [1507 1218] [1273 1489]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=2/sel=false
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=2/sel=false (0.00s)
            sort_test.go:186: for input [[1740 1699] [129 1164] [1507 1218] [817 1127] [1550 189] [207 754] [1273 1489] [1273 274]]:
                expected [[1740 1699] [1550 189] [1507 1218] [1273 274]], actual [[1740 1699] [1550 189] [1507 1218] [1273 1489]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=16/sel=true
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=16/sel=true (0.00s)
            sort_test.go:186: for input [[1740 1699] [129 1164] [1507 1218] [817 1127] [1550 189] [207 754] [1273 1489] [1273 274]]:
                expected [[1740 1699] [1550 189] [1507 1218] [1273 274]], actual [[1740 1699] [1550 189] [1507 1218] [1273 1489]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true
    --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true (0.00s)

```

SHA: https://github.com/cockroachdb/cockroach/commits/056afc72397974ed9489bf11f16ac8e2fc281a67

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestSortRandomized PKG=github.com/cockroachdb/cockroach/pkg/sql/exec TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1342399&tab=buildLog

```
            sort_test.go:186: for input [[1609 670 1835] [394 646 1940] [1506 646 1047] [808 1607 244] [134 1624 1038] [569 809 310] [1638 578 1981] [1385 1739 1377]]:
                expected [[1638 578 1981] [394 646 1940] [1506 646 1047] [1609 670 1835]], actual [[1638 578 1981] [1506 646 1047] [394 646 1940] [1609 670 1835]]
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=3/sel=true
        --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=3/sel=true (0.00s)
            sort_test.go:186: for input [[1609 670 1835] [394 646 1940] [1506 646 1047] [808 1607 244] [134 1624 1038] [569 809 310] [1638 578 1981] [1385 1739 1377]]:
                expected [[1638 578 1981] [394 646 1940] [1506 646 1047] [1609 670 1835]], actual [[1638 578 1981] [1506 646 1047] [394 646 1940] [1609 670 1835]]
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=16/sel=false
        --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=16/sel=false (0.00s)
            sort_test.go:186: for input [[1609 670 1835] [394 646 1940] [1506 646 1047] [808 1607 244] [134 1624 1038] [569 809 310] [1638 578 1981] [1385 1739 1377]]:
                expected [[1638 578 1981] [394 646 1940] [1506 646 1047] [1609 670 1835]], actual [[1638 578 1981] [1506 646 1047] [394 646 1940] [1609 670 1835]]
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=1024/sel=false
        --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=1024/sel=false (0.00s)
            sort_test.go:186: for input [[1609 670 1835] [394 646 1940] [1506 646 1047] [808 1607 244] [134 1624 1038] [569 809 310] [1638 578 1981] [1385 1739 1377]]:
                expected [[1638 578 1981] [394 646 1940] [1506 646 1047] [1609 670 1835]], actual [[1638 578 1981] [1506 646 1047] [394 646 1940] [1609 670 1835]]
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=1024/sel=true
        --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=1024/sel=true (0.00s)
            sort_test.go:186: for input [[1609 670 1835] [394 646 1940] [1506 646 1047] [808 1607 244] [134 1624 1038] [569 809 310] [1638 578 1981] [1385 1739 1377]]:
                expected [[1638 578 1981] [394 646 1940] [1506 646 1047] [1609 670 1835]], actual [[1638 578 1981] [1506 646 1047] [394 646 1940] [1609 670 1835]]
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true
    --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true (0.00s)
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=1/sel=false
        --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=1/sel=false (0.00s)
            sort_test.go:186: for input [[1609 670 1835] [394 646 1940] [1506 646 1047] [808 1607 244] [134 1624 1038] [569 809 310] [1638 578 1981] [1385 1739 1377]]:
                expected [[1638 578 1981] [394 646 1940] [1506 646 1047] [1609 670 1835]], actual [[1638 578 1981] [1506 646 1047] [394 646 1940] [1609 670 1835]]
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=2/sel=true
        --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=2/sel=true (0.00s)
            sort_test.go:186: for input [[1609 670 1835] [394 646 1940] [1506 646 1047] [808 1607 244] [134 1624 1038] [569 809 310] [1638 578 1981] [1385 1739 1377]]:
                expected [[1638 578 1981] [394 646 1940] [1506 646 1047] [1609 670 1835]], actual [[1638 578 1981] [1506 646 1047] [394 646 1940] [1609 670 1835]]
=== RUN   TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=3/sel=false
        --- FAIL: TestSortRandomized/nCols=3/nOrderingCols=1/topK=true/batchSize=3/sel=false (0.00s)
            sort_test.go:186: for input [[1609 670 1835] [394 646 1940] [1506 646 1047] [808 1607 244] [134 1624 1038] [569 809 310] [1638 578 1981] [1385 1739 1377]]:
                expected [[1638 578 1981] [394 646 1940] [1506 646 1047] [1609 670 1835]], actual [[1638 578 1981] [1506 646 1047] [394 646 1940] [1609 670 1835]]

```

SHA: https://github.com/cockroachdb/cockroach/commits/056afc72397974ed9489bf11f16ac8e2fc281a67

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestSortRandomized PKG=github.com/cockroachdb/cockroach/pkg/sql/exec TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1342402&tab=buildLog

```
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true
    --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true (0.01s)
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=2/sel=false
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=2/sel=false (0.00s)
            sort_test.go:186: for input [[1795 868] [689 1233] [1609 1789] [1022 496] [1139 915] [1579 868] [1877 668] [659 1962]]:
                expected [[1022 496] [1877 668] [1579 868] [1795 868]], actual [[1022 496] [1877 668] [1795 868] [1579 868]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=1/sel=true
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=1/sel=true (0.00s)
            sort_test.go:186: for input [[1795 868] [689 1233] [1609 1789] [1022 496] [1139 915] [1579 868] [1877 668] [659 1962]]:
                expected [[1022 496] [1877 668] [1579 868] [1795 868]], actual [[1022 496] [1877 668] [1795 868] [1579 868]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=3/sel=false
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=3/sel=false (0.00s)
            sort_test.go:186: for input [[1795 868] [689 1233] [1609 1789] [1022 496] [1139 915] [1579 868] [1877 668] [659 1962]]:
                expected [[1022 496] [1877 668] [1579 868] [1795 868]], actual [[1022 496] [1877 668] [1795 868] [1579 868]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=3/sel=true
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=3/sel=true (0.00s)
            sort_test.go:186: for input [[1795 868] [689 1233] [1609 1789] [1022 496] [1139 915] [1579 868] [1877 668] [659 1962]]:
                expected [[1022 496] [1877 668] [1579 868] [1795 868]], actual [[1022 496] [1877 668] [1795 868] [1579 868]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=16/sel=false
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=16/sel=false (0.00s)
            sort_test.go:186: for input [[1795 868] [689 1233] [1609 1789] [1022 496] [1139 915] [1579 868] [1877 668] [659 1962]]:
                expected [[1022 496] [1877 668] [1579 868] [1795 868]], actual [[1022 496] [1877 668] [1795 868] [1579 868]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=16/sel=true
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=16/sel=true (0.00s)
            sort_test.go:186: for input [[1795 868] [689 1233] [1609 1789] [1022 496] [1139 915] [1579 868] [1877 668] [659 1962]]:
                expected [[1022 496] [1877 668] [1579 868] [1795 868]], actual [[1022 496] [1877 668] [1795 868] [1579 868]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=1024/sel=false
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=1024/sel=false (0.00s)
            sort_test.go:186: for input [[1795 868] [689 1233] [1609 1789] [1022 496] [1139 915] [1579 868] [1877 668] [659 1962]]:
                expected [[1022 496] [1877 668] [1579 868] [1795 868]], actual [[1022 496] [1877 668] [1795 868] [1579 868]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=1024/sel=true
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=1024/sel=true (0.00s)
            sort_test.go:186: for input [[1795 868] [689 1233] [1609 1789] [1022 496] [1139 915] [1579 868] [1877 668] [659 1962]]:
                expected [[1022 496] [1877 668] [1579 868] [1795 868]], actual [[1022 496] [1877 668] [1795 868] [1579 868]]
=== RUN   TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=1/sel=false
        --- FAIL: TestSortRandomized/nCols=2/nOrderingCols=1/topK=true/batchSize=1/sel=false (0.00s)
            sort_test.go:186: for input [[1795 868] [689 1233] [1609 1789] [1022 496] [1139 915] [1579 868] [1877 668] [659 1962]]:
                expected [[1022 496] [1877 668] [1579 868] [1795 868]], actual [[1022 496] [1877 668] [1795 868] [1579 868]]

```

--- a/pkg/sql/exec/sort_test.go
+++ b/pkg/sql/exec/sort_test.go
@@ -156,6 +156,9 @@ func TestSortRandomized(t *testing.T) {
 							// Small range so we can test partitioning
 							tups[i][j] = rng.Int63() % 2048
 						}
+						// Enforce that the last ordering column is always unique. Otherwise
+						// there would be multiple valid sort orders.
+						tups[i][ordCols[nOrderingCols-1].ColIdx] = int64(i)
 					}
 
 					expected := make(tuples, nTups)

